sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java

```diff
@@ -0,0 +1,1064 @@
+            "jsr305.contradicting.return.value.annotations";
+
+    /** Key for error message. */
+    public static final String MSG_OVERRIDDEN_METHOD_WITH_CHECK_RETURN_VALUE =
+            "jsr305.overridden.method.with.check.return.value.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_REDUNDANT_NONNULL_BY_DEFAULT_ANNOTATION =
+            "jsr305.redundant.nonnull.by.default.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_REDUNDANT_NULLABLE_BY_DEFAULT_ANNOTATION =
+            "jsr305.redundant.nullable.by.default.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_VOID_WITH_CHECK_RETURN_VALUE_ANNOTATION =
+            "jsr305.void.with.check.return.value.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_REDUNDANT_NONNULL_RETURN_ANNOTATION =
+            "jsr305.redundant.nonnull.return.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_RETURN_WITHOUT_NULLNESS_ANNOTATION =
+            "jsr305.return.without.nullness.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_OVERRIDDEN_METHODS_ALLOW_ONLY_NONNULL =
+            "jsr305.overridden.methods.allow.only.nonnull";
+
+    /** Key for error message. */
+    public static final String MSG_NEED_TO_INHERIT_PARAM_ANNOTATIONS =
+            "jsr305.need.to.inherit.param.annotations";
+
+    /** Key for error message. */
+    public static final String MSG_CONSTRUCTOR_WITH_RETURN_ANNOTATION =
+            "jsr305.constructor.with.return.annotation";
+
+    /** Package name. */
+    private static final String PKG_JAVAX_ANNOTATION = "javax.annotation";
+
+    /**
+     * Class NullnessAnnotation. The annotations we consider as "nullness-relevant".
+     */
+    protected enum NullnessAnnotation {
+
+        /** Override. */
+        OVERRIDE("Override", "java.lang"),
+        /** CheckForNull. */
+        CHECK_FOR_NULL("CheckForNull", PKG_JAVAX_ANNOTATION),
+        /** Nullable. */
+        NULLABLE("Nullable", PKG_JAVAX_ANNOTATION),
+        /** Nonnull. */
+        NONNULL("Nonnull", PKG_JAVAX_ANNOTATION),
+        /** CheckReturnValue. */
+        CHECK_RETURN_VALUE("CheckReturnValue", PKG_JAVAX_ANNOTATION),
+        /** ParametersAreNonnullByDefault. */
+        PARAMETERS_ARE_NONNULL_BY_DEFAULT("ParametersAreNonnullByDefault", PKG_JAVAX_ANNOTATION),
+        /** ParametersAreNullableByDefault. */
+        PARAMETERS_ARE_NULLABLE_BY_DEFAULT("ParametersAreNullableByDefault", PKG_JAVAX_ANNOTATION),
+        /** ReturnValuesAreNonnullByDefault. */
+        RETURN_VALUES_ARE_NONNULL_BY_DEFAULT("ReturnValuesAreNonnullByDefault",
+                "edu.umd.cs.findbugs.annotations");
+
+        /** The annotation's name. */
+        private final String theAnnotationName;
+
+        /** The annotation's fully qualified class name. */
+        private final String theFqcn;
```

Comment:
fullyQualifiedClassName

lets not economize on names, especially if that force creation of comment.

Reply:
Done with next commit.